pyscf/dft/rks.py

```diff
@@ -82,14 +100,28 @@ def get_veff(ks, mol=None, dm=None, dm_last=0, vhf_last=0, hermi=1):
             getattr(vhf_last, 'vk', None) is not None):
             ddm = numpy.asarray(dm) - numpy.asarray(dm_last)
             vj, vk = ks.get_jk(mol, ddm, hermi)
+            vk *= hyb
+            if abs(omega) > 1e-10:
+                mol.set_range_coulomb(omega)
+                vklr = jk.get_jk(mol, ddm, 'ijkl,jk->il')
+                vklr *= (alpha - hyb)
+                vk += vklr
+                mol.set_range_coulomb(0)
```

Comment:
I'm confused. Why do you call get_jk, when the only thing you need is K?

Reply:
Also, I think it would be faster to evaluate the **short-range** exchange, instead of the **long-range** one, if you're doing any screening.

Reply:
There is no get_k yet.
PySCF only has erf integrals at the moment.

Reply:
get_k function does not call the range separated coulomb operator when integrals are cached in memory.

Reply:
I'm not sure I understand what you mean: would calling  ```get_k()``` give you the wrong answer then?

Does ```mol``` store the integrals? Shouldn't you have different objects to store the integrals for different values of omega?

Reply:
integrals are cached in mf object if they can be fit in memory.  I didn't consider the possibility of multiple types of coulomb integrals when I wrote the scf class.  mf.get_k function will use the cached integrals to construct K matrix no matter which omega value is assigned.  jk.get_jk function computes the JK matrix directly.